ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,223 @@
+            res.send(_result(400, 'encryptionKeyId format wrong'))
+            return
+        }
+        if (!check_rotationInterval_format(rotationInterval)) {
+            res.send(_result(400, 'rotationInterval format wrong'))
+            return
+        }
+
+        const secret_name_query = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['appid', 'secretName'],
+            limit: 1,
+        }
+        let secret_name_result = await DB.partitionedFind('secret_metadata', secret_name_query)
+        if (secret_name_result.docs.length > 0) {
+            res.send(_result(400, 'Secret name already exists, cannot use the same secret name'))
+            return
+        }
+
+        if (encryptionKeyId != '' && typeof (encryptionKeyId) == 'string') {
+            let key_id_res = await DB.partitionedFind('cmk', queryKeyId(appid, encryptionKeyId))
+            if (key_id_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find key id'))
+                return
+            } else {
+                keyBlob = key_id_res.docs[0].keyBlob
+            }
+        } else if (encryptionKeyId == "" || encryptionKeyId == undefined) {
+            let default_cmk_res = await DB.partitionedFind('user_info', queryDefaultCMK(appid))
+            if (default_cmk_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find default CMK'))
+                return
+            }
+            keyBlob = default_cmk_res.docs[0].sm_default_cmk
+        } else {
+            res.send(_result(400, 'encryptionKeyId error'))
+            return
+        }
+        const apikey_encrypt_res = napi_result(cryptographic_apis.Encrypt, res, [keyBlob, secretData, ''])
+        const { ciphertext } = apikey_encrypt_res.result
+        if (rotationInterval != '' && rotationInterval != undefined) {
+            let timenum = rotationInterval.substring(0, rotationInterval.length - 1)
+            let unit = rotationInterval.substring(rotationInterval.length - 1, rotationInterval.length)
+            switch (unit) {
+                case (unit = 'd'):
+                    nextRotationDate = timenum * 24 * 60 * 60 * 1000 + createTime
+                    break
+                case (unit = 'h'):
+                    nextRotationDate = timenum * 60 * 60 * 1000 + createTime
+                    break
+                case (unit = 'm'):
+                    nextRotationDate = timenum * 60 * 1000 + createTime
+                    break
+                case (unit = 's'):
+                    nextRotationDate = timenum * 1000 + createTime
+                    break
+            }
+            let interval = nextRotationDate - createTime
+
+            console.info('interval', interval)
+            if (6 * 60 * 60 * 1000 > interval || interval > 8760 * 60 * 60 * 1000) {
+                res.send(_result(400, 'rotationInterval is too short'))
```

Comment:
the error msg is not make sense.

Reply:
it will change to 'The rotation time is less than 6 hours or more than 8760 hours'